### PR TITLE
Update base image to Alma Linux 9

### PR DESCRIPTION
**Pull request by ebocchi**

### CODE REVIEW

**Comment by rptaylor** [comment]
@ebocchi thanks for starting this! I can deploy and test this but we need the image to be built first. 
If merged I guess this will currently overwrite the image used in production `gitlab-registry.cern.ch/sciencebox/docker-images/frontier-squid:4.17-2.1`.  

Do you think it would make sense to merge with BASE_TAG updated to `'gitlab-registry.cern.ch/sciencebox/docker-images/frontier-squid-el9'` to make a separate EL9 image for testing purposes?

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM cern/cc7-base:20221101-1
+FROM cern/alma9-base:20240501-1
 
 MAINTAINER Enrico Bocchi <enrico.bocchi@cern.ch>
```
Comment (rptaylor) [inline Dockerfile:4]: All 'yum' should be replaced by 'dnf', except for /etc/yum.repos.d

**Comment by rptaylor** [comment]
As a separate/independent next step I started https://github.com/sciencebox/charts/pull/124  to prepare the chart for a squid update to v5.

**Comment by ebocchi** [comment]
> @ebocchi thanks for starting this! I can deploy and test this but we need the image to be built first. If merged I guess this will currently overwrite the image used in production `gitlab-registry.cern.ch/sciencebox/docker-images/frontier-squid:4.17-2.1`.
> 
> Do you think it would make sense to merge with BASE_TAG updated to `'gitlab-registry.cern.ch/sciencebox/docker-images/frontier-squid-el9'` to make a separate EL9 image for testing purposes?

Tagged as `gitlab-registry.cern.ch/sciencebox/docker-images/frontier-squid:el9-4.17-2.1`

**Comment by rptaylor** [comment]
Thanks @ebocchi , I tried it on a dev cluster, and ran cvmfs_config chksetup and ALRB db-fnget tests on the squids.
I did not see any problems. I think it should be okay to use in production.

Do you think it makes sense to merge with the same `4.17-2.1` tag so that it will start to be used transparently? I'm okay with whatever would be easiest.

**Comment by rptaylor** [comment]
@ebocchi any further thoughts on this?  I think it's okay to merge, however you want to handle the tag version is fine with me.

**Comment by rptaylor** [comment]
I built and pushed a new image to `gitlab-registry.cern.ch/sciencebox/docker-images/frontier-squid:4.17-2.1.el9`

**Comment by rptaylor** [comment]
I deployed this image using chart frontier-squid-0.1.7 on our dev cluster, and run the ALRB db-fnget test and cvmfs_config chksetup on a node, all looked fine.

Squid pod logs and `squidclient mgr:info` also look fine.

**Comment by rptaylor** [comment]
Summary of changes:
- update to base image cern/alma9-base:20240701-2
- update Dockerfile to use DNF
- update version tag to have `.el9` at the end

**Pull request merged**